7a_temp_coop_munge/src/data_parsers/parse_Iowa_DNR_LimnoProfiles_2000_2020.R

```diff
@@ -0,0 +1,507 @@
+                                          htm_data_start = 3,
+                                          htm_verbose = F)) %>%
+    dplyr::mutate(date = as.Date(sampling_date_time),
+                  time = format(sampling_date_time, '%H:%M'),
+                  Timezone = NA) %>%
+    dplyr::select(date, time, Timezone, depth, temp, lakeid)
+
+  # Clean data from 2017a and 2017b----------------------------------------------
+  #' 2017a data does not have date or time
+  #' 2017b data does have date and time. Lake name needs to be parsed out of file.
+
+  dat_2017b <- files_2017b %>%
+    purrr::map_dfr(~ parse_2017_2020_data(.,
+                                          keep_cols = c('DATE', 'TIME', 'SITE',
+                                                        'Temp...F.', 'Depth..m.'),
+                                          lakeid_col = 3,
+                                          temp_as_f = T,
+                                          use_readr = F)) %>%
+    dplyr::mutate(Timezone = NA) %>%
+    dplyr::rename(depth = depth..m.,
+           temp = temp...f.) %>%
+    dplyr::select(date, time, Timezone, depth, temp, lakeid)
+
+  # Clean data from 2018-2020 ---------------------------------------------------
+  ## `2019_R3_Profiles.csv` is in an unexpected format. it was faster to just munge
+  ## separately
+
+  dat_2018_2020_1 <- files_2018_2020[c(1:5, 7:9)] %>%
+    purrr::map_dfr(~ parse_2017_2020_data(.,
+                                          keep_cols = c('Lake', 'Date', 'Time',
+                                                        'Depth (m)', 'Temperature (Deg C)'),
+                                          lakeid_col = 1,
+                                          temp_as_f = F,
+                                          use_readr = T)) %>%
+    dplyr::rename(depth = `depth (m)`,
+           temp = `temperature (deg c)`)
+
+
+  dat_2018_2020_2 <- files_2018_2020[6] %>%
+    purrr::map_dfr(~ parse_2017_2020_data(.,
+                                          keep_cols = c('DATE', 'DATE.1', 'TIME',
+                                                        'Depth..m.', 'Temperature..Deg.C.'),
+                                          lakeid_col = 1,
+                                          temp_as_f = F,
+                                          use_readr = F)) %>%
+    dplyr::rename(depth = depth..m.,
+           temp = temperature..deg.c.)
+
+  # clean data -------------------
+  data_clean <- dplyr::bind_rows(dat_2000_2003,
+                          dat_2004_2007,
+                          dat_2009,
+                          dat_2010,
+                          dat_2011_2016,
+                          dat_2017b,
+                          dat_2018_2020_1,
+                          dat_2018_2020_2) %>%
+    dplyr::mutate(Iowa_ID = paste('Iowa_', lakeid, sep = ''), # for matching the xwalk
+      DateTime = as.Date(date)) %>% # there is a mix of POSIXct and Date in the indiv dfs
```

Comment:
Interesting...if `date` has different classes, I'd assume the `bind_rows` would either fail or coerce them all into the same. E.g.,
```r
bind_rows(tibble(data = 2), tibble(data = "4"))
Error: Can't combine `..1$data` <double> and `..2$data` <character>.
```
Similar to my comment above on temperature as °C or °F, it seems each of the individual parsers would be the place to ensure a common unit/class for what is coming out of it, so that the bind is simple (I do, however, agree that the `Iowa_ID` should happen here once, which makes it more future-proof if we for some name modify that ID name in the future, we'd only have to change it here for the parsers).

Reply:
I *think* it works because the two classes are `Date` and `POSIXct` and they're close enough (i.e., `dplyr::bind_rows` can handle two similar dates by coercing one to another). I did it at the end as a kludge because the pipeline was breaking when binding the dates. 

As you noted, the most correct thing to do would be to clean it up within the individual parsers, which I will.